spec/shared/room_spec.rb

```diff
@@ -0,0 +1,383 @@
+shared_examples 'room_behavior' do |room_type, query:|
```

Comment:
It would also be good to see the `room_type` parameter turned into a keyword argument. Passing through 'p' or 'c' doesn't make it particularly readable. 
```
shared_examples 'room_behavior' do |room_type: nil, query: false|
```
???